messaging/src/test/java/org/axonframework/commandhandling/gateway/DefaultCommandGatewayTest.java

```diff
@@ -77,6 +85,41 @@ void dispatchReturnsExceptionallyCompletedFutureWhenCommandBusReturnsExceptional
         assertTrue(result.getResultMessage().isCompletedExceptionally());
     }
 
+    @Test
+    void resolvesQualifiedNameUsingMessageNameResolver() throws ExecutionException, InterruptedException {
+        // given
+        when(mockCommandBus.dispatch(any(), any())).thenAnswer(i -> CompletableFuture.completedFuture(
+                new GenericCommandResultMessage<>(new QualifiedName("test", "result", "0.0.1"), "OK")
+        ));
+
+        // when
+        TestPayload payload = new TestPayload();
+        CommandResult result = testSubject.send(payload, null);
+
+        // then
+        var expectedQualifiedName = new QualifiedName("test", "TestPayload", "0.5.0");
+        verify(mockCommandBus).dispatch(argThat(m -> m.name().equals(expectedQualifiedName)), isNull());
+        assertEquals("OK", result.getResultMessage().get().getPayload());
+    }
+
+    @Test
+    void passCommandMessageAsIs() throws ExecutionException, InterruptedException {
+        // given
+        when(mockCommandBus.dispatch(any(), any())).thenAnswer(i -> CompletableFuture.completedFuture(
+                new GenericCommandResultMessage<>(new QualifiedName("test", "result", "0.0.1"), "OK")
+        ));
+
+        // when
+        TestPayload payload = new TestPayload();
+        var testCommand =
+                new GenericCommandMessage<>(new QualifiedName("test", "command", "1.0.0"), payload);
+        CommandResult result = testSubject.send(testCommand, null);
+
+        // then
+        verify(mockCommandBus).dispatch(argThat(m -> m.equals(testCommand)), isNull());
+        assertEquals("OK", result.getResultMessage().get().getPayload());
+    }
```

Comment:
Test to be sure that we keep the old behaviour with support for Message implementation as `CommandGateway.send` argument.

Reply:
For now, will be dropped in new PRs.